### PR TITLE
Params Collections: Implement language version check.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@RikkiGibson, @333fred, @dotnet/roslyn-compiler Please review. This is a fairly small change

File: src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs [hunk 1]
```diff
@@ -444,6 +578,87 @@ static void Test(params IEnumerable a)
...
+                );
+        }
+
+        [Fact]
+        public void LanguageVersion_02_CallSite()
+        {
+            var src1 = @"
+public class Params
+{
+    static public void Test1(params System.ReadOnlySpan<long> a) {}
```
Comment (333fred) [inline src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs:617]: Do we carry params-ness through to delegate natural types? Consider adding some variants of this test for delegate conversions as well, even if we don't, just to verify that behavior.

Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs:617]: > Do we carry params-ness through to delegate natural types? 

At the moment this is done only for arrays. Expanding the behavior to other types is on my list to do.

> Consider adding some variants of this test for delegate conversions as well,

I prefer to do that once "carry-over" part is implemented. But this is definitely a good scenario to cover.

Comment (333fred) [reply src/Compilers/CSharp/Test/Emit2/Semantics/ParamsCollectionTests.cs:617]: > At the moment this is done only for arrays. Expanding the behavior to other types is on my list to do.

Gotcha, thanks.

**Review by 333fred — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@RikkiGibson, @dotnet/roslyn-compiler For the second review. This is a fairly small change

**Review by RikkiGibson — APPROVED** [review]

**Pull request merged**